pkg/multicluster/auth/interfaces.go

```diff
@@ -0,0 +1,66 @@
+package auth
+
+import (
+	"context"
+
+	k8s_core_v1 "github.com/solo-io/skv2/pkg/generated/kubernetes/core/v1"
+	k8s_core_types "k8s.io/api/core/v1"
+	k8s_rbac_types "k8s.io/api/rbac/v1"
+	"k8s.io/client-go/rest"
+)
+
+//go:generate mockgen -source ./interfaces.go -destination ./mocks/mock_auth.go
+
+type RbacBinder interface {
```

Comment:
Hard -1 to letting this live in two separate places. There's ongoing churn in this section of code and I do not want to maintain it in two separate repos.

I'm not going to approve this until we have a clear ownership/path to integrating it back into SMH

Reply:
To be clear: By "this" I mean the entirety of this PR, not this file in particular